### PR TITLE
use ArrayListMultimap instead of new java.util.HashMap[InternalRow, j…

### PR DESCRIPTION
**Pull request by LuciferYang**
ArrayListMultimap is a HashMap associates each key with an ArrayList of values.

### CODE REVIEW

**Comment by linhong-intel** [comment]
do you have some performance comparison to show there is a improvement.

File: src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/BTreeIndexWriter.scala [hunk 1]
```diff
@@ -22,6 +22,7 @@ import java.util.Comparator
 
 import scala.collection.JavaConverters._
 
+import com.google.common.collect.ArrayListMultimap
```
Comment (adrian-wang) [inline src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/BTreeIndexWriter.scala]: Nit: add a new line behind this.
the import section will divide into 4 parts. java - scala - third party - spark/oap.

Comment (LuciferYang) [inline src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/BTreeIndexWriter.scala]: ok

Comment (LuciferYang) [inline src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/BTreeIndexWriter.scala]: but why...,guava&hadoop not belong to same group?

Comment (adrian-wang) [inline src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/BTreeIndexWriter.scala]: ah... sorry, you are right.

**Comment by adrian-wang** [comment]
LGTM then.

**Comment by adrian-wang** [comment]
@linhong-intel I took a look at the implementation of `ArrayListMultimap`, it is similar as the way our code is. So I think the performance would be similar too, but code looks simplified.
I'm merging this. Thanks @LuciferYang

**Pull request merged**